core/ustring.cpp

```diff
@@ -1267,7 +1265,7 @@ String String::num_real(double p_num) {
 
 	if ((int)p_num != p_num) {
 
-		double dec = p_num - (float)((int)p_num);
+		double dec = p_num - (double)((int)p_num);
 
 		int digit = 0;
 		int decimals = MAX_DIGITS;
```

Comment:
Since this method is called `num_real`, I think it should reflect the precision of `real_t` - the values should be kept as double, but we should change this to use less digits (the previous value of `MAX_DIGITS`) unless `REAL_T_IS_DOUBLE`.

```
#if REAL_T_IS_DOUBLE
		int decimals = MAX_DIGITS;
#else
		int decimals = 6;
#endif
```